src/test/java/marquez/api/NamespaceIntegrationTest.java

```diff
@@ -0,0 +1,231 @@
+    assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), res.getStatus());
+  }
+
+  @Test
+  public void testGetAllNamespaceWithEmptyResultSet() {
+    final Response res =
+        APP.client()
+            .target(URI.create("http://localhost:" + APP.getLocalPort()))
+            .path("/namespaces/")
+            .request(MediaType.APPLICATION_JSON)
+            .get();
+    assertEquals(Response.Status.OK.getStatusCode(), res.getStatus());
+
+    GetAllNamespacesResponse responseBody = res.readEntity(GetAllNamespacesResponse.class);
+    List<Namespace> returnedList = responseBody.getNamespaces();
+
+    assertThat(returnedList.isEmpty());
+  }
+
+  @Test
+  public void testGetAllNamespaceWithSingleResultSet() throws NamespaceException {
+    final List<Namespace> existingNamespaces = Collections.singletonList(TEST_NAMESPACE);
+    NamespaceService namespaceService = Mockito.mock(NamespaceService.class);
+    NamespaceResource namespaceResource = new NamespaceResource(namespaceService);
+
+    when(namespaceService.getAllNamespaces()).thenReturn(existingNamespaces);
+    Response res = namespaceResource.getAllNamespaces();
+
+    GetAllNamespacesResponse responseBody = (GetAllNamespacesResponse) res.getEntity();
+    List<Namespace> returnedList = responseBody.getNamespaces();
+
+    assertThat(returnedList).contains(TEST_NAMESPACE);
+  }
+
+  @Test
+  public void testGetAllNamespaceWithMultipleResultSet() throws NamespaceException {
+    final List<Namespace> existingNamespaces = new ArrayList<>();
+    Namespace secondNamespace =
+        new Namespace(
+            UUID.randomUUID(),
+            Timestamp.from(Instant.now()),
+            "someOtherName",
+            "someOtherOwner",
+            "a second ns for testing");
+    existingNamespaces.add(TEST_NAMESPACE);
+    existingNamespaces.add(secondNamespace);
+
+    NamespaceService namespaceService = Mockito.mock(NamespaceService.class);
+    NamespaceResource namespaceResource = new NamespaceResource(namespaceService);
+
+    when(namespaceService.getAllNamespaces()).thenReturn(existingNamespaces);
+    Response res = namespaceResource.getAllNamespaces();
+
+    GetAllNamespacesResponse responseBody = (GetAllNamespacesResponse) res.getEntity();
+    List<Namespace> returnedList = responseBody.getNamespaces();
+
+    assertThat(returnedList).containsExactly(TEST_NAMESPACE, secondNamespace);
+  }
+
+  @Test
+  public void testGetAllNamespacesErrorHandling() throws NamespaceException {
+    NamespaceService namespaceService = Mockito.mock(NamespaceService.class);
+    NamespaceResource namespaceResource = new NamespaceResource(namespaceService);
+    when(namespaceService.getAllNamespaces()).thenThrow(NamespaceException.class);
+
+    Response res = namespaceResource.getAllNamespaces();
+    assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), res.getStatus());
+  }
+
+  @Test
+  public void testGetAllNamespacesWithoutTrailingSlash() {
```

Comment:
is this test needed? we won't ever change how the trailing slash is treated, it's really just testing how dropwizard works

Reply:
Good point. Will remove.